data-products/src/main/scala/org/sunbird/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -0,0 +1,280 @@
+      "/tmp/druid-reports"), config.get("accountKey").asInstanceOf[Option[String]]);
+    val format = config.get("format").get.asInstanceOf[String]
+    val reportId = config.get("reportId").get.asInstanceOf[String]
+    val quoteColumns =config.get("quoteColumns").getOrElse(List()).asInstanceOf[List[String]]
+    var duplicateDimsDf = data
+    if(quoteColumns.nonEmpty) {
+      import org.apache.spark.sql.functions.udf
+      val quoteStr = udf((column: String) =>  "\'"+column+"\'")
+      quoteColumns.map(column => {
+        duplicateDimsDf = duplicateDimsDf.withColumn(column, quoteStr(col(column)))
+      })
+    }
+    val deltaFiles = duplicateDimsDf.saveToBlobStore(storageConfig, format, reportId, Option(Map("header" -> "true")), None)
+    deltaFiles
+  }
+
+  def processRequest(request: JobRequest, reportConfig:ReportConfig, storageConfig:StorageConfig)(implicit spark: SparkSession, fc: FrameworkContext, sqlContext:SQLContext, sc:SparkContext,config: JobConfig,conf: Configuration): JobRequest ={
+    markRequestAsProcessing(request)
+    val requestBody = JSONUtils.deserialize[RequestBody](request.request_data)
+    val requestParamsBody = requestBody.`params`
+    reportConfig.metrics.map(metric => {
+      metric.druidQuery.filters.get.map(filt=>{
+        (requestBody.`params`.keys).map(ke=> {
+          if(filt.get("value").contains("$"+ke)) {
+            val dynamicFilterValue = requestParamsBody.get(ke).get.toString
+            filt.update("value", dynamicFilterValue)
+          }
+        })
+      })
+    })
+    reportConfig.output.foreach({ot => {
+      ot.update("label",((System.currentTimeMillis())+"_"+request.request_id).toString)
+    } })
+    val druidData : RDD[DruidOutput] = druidAlgorithm(reportConfig)
+    val result = CommonUtil.time(druidPostProcess(druidData,JSONUtils.deserialize[ReportConfigImmutable](JSONUtils.serialize(reportConfig)),storageConfig))
+    val response = result._2;
+    val failedOnDemandDruidRes = response.status.equals("FAILED")
+    if (failedOnDemandDruidRes) {
+      markRequestAsFailed(request, response.statusMsg)
+    } else {
+      if(validateEncryptionRequest(request) == true){
+        val storageConfig = getStorageConfig(config, response.file.head)
+        request.download_urls = Option(response.file);
+        request.execution_time = Option(result._1);
+        processRequestEncryption(storageConfig,request)
+        request.status = "SUCCESS";
+        request.dt_job_completed = Option(System.currentTimeMillis)
+      }
+      else {
+        request.status = "SUCCESS";
+        request.download_urls = Option(response.file);
+        request.execution_time = Option(result._1);
+        request.dt_job_completed = Option(System.currentTimeMillis)
+      }
+    }
+    request
+  }
+  def saveRequestAsync(request: JobRequest,jobId:String)(implicit conf: Configuration, fc: FrameworkContext): CompletableFuture[JobRequest] = {
```

Comment:
Same with saveRequestAsync - this method already available in OnDemandExhaustJob - why is this implemented again here?

Reply:
As discussed with you , we renamed the method with proper naming convention